tools/install.sh

```diff
@@ -6,15 +6,16 @@ if [ -z "${BASH_VERSION-}" ]; then
   printf "Error: Bash 3.2 or higher is required for Oh My Bash.\n"
   printf "Error: Install Bash and try running this installation script with Bash.\n"
   if command -v bash >/dev/null 2>&1; then
+    # shellcheck disable=SC2016
     printf 'Example: \033[31;1mbash\033[0;34m -c "$(curl -fsSL https://raw.githubusercontent.com/ohmybash/oh-my-bash/master/tools/install.sh)"\n'
   fi
-  return 1 >/dev/null 2>&1 || exit 1
+  return 1 >/dev/null 2>&1
```

Comment:
This is the idiom when the file can be used as either of `source tools/install.sh` or `bash tools/install.sh`.

Reply:
Thanks for teaching me that! At first, I was confused because in the [Bash manual](https://www.gnu.org/software/bash/manual/bash.html#index-return-status) it was stated that `return` was a synonym for `exit`, so I thought that creating this compound command with the `OR` conditional was redundant. However, I tried doing `return 1` in my Bash shell to see what would happen, and it indeed said `-bash: return: can only `return' from a function or sourced script`. I was still befuddled, so I made a script that simply called `return 1` and invoked it with `bash` only to get this same error. When I changed the script to run `return 1 || exit 1` and sourced it again, it produced nothing on standard error and finished with exit code `1`. It seems rather quirky that `exit` and `return` would have different semantics despite being called synonymous, but this is definitely something I'm glad to have learned.

In any case, there should now be ShellCheck ignore lines in all places where this change should be applicable.

Reply:
That section of the Bash manual defines the general terms used in the text but not in the code. The *return status* and the *exit status* do not represent `return <status>` and `exit <status>` in the code, but the words in the textual explanation like "the *return status* of a coprocess is ...." and "the *exit status* of a pipe line is ...".

> It seems rather quirky that `exit` and `return` would have different semantics despite being called synonymous,

So, they are not synonyms. What are the synonyms are the words *return status* and *exit status*.

> In any case, there should now be ShellCheck ignore lines in all places where this change should be applicable.

Hmm, my `shellcheck` doesn't seem to complain about `return 2 &>/dev/null || exit 2`. Maybe this is related to the version of `shellcheck`. My version is 0.8.0, but the latest version seems to be 0.9.0. I have checked the `shellcheck` behavior of the online editor at https://www.shellcheck.net/ which seems to complain about it.

I thought it might be worth reporting the issue, but I found an existing report for this false positive: https://github.com/koalaman/shellcheck/issues/2620.

Anyway, it is good to keep `disable=SC2317` for a while even if it would be fixed at the shellcheck's side.